docs/howto/deploy-subgraph/deploying-subgraph.md

```diff
@@ -0,0 +1,995 @@
+
+## 1 What is The graph
+
+For an NFT or gaming application, we can find directly the owner of a NFT or an item based on their ID.  
+For doing this, we only have to request a read operation on the contract.  
+But more advanced queries like finding all NFT or items associated to a user is not possible only by querying the read method's on contract.  
+For doing this we would have to process every transfer event emmitted, agregate data and process.
+This is not possible for a dapp only running on a browser.  
+With a server thinks are not much better, because this require a centralized cache to store data and thus it introduce a security weakness.  
+That is why The Graph network was created.
+
+The graph is an opensource software based on a decentralized protocol that work as a cache  
+for indexing and querying data from blockchain.  
+It was initially created for ethereum blockchain and then started to integrate other blockchain. 
+Were as a classic database, on blockchains you cannot perform a query on data by criteria.
+
+The Graph retrieves blocks and extract data before storing it into indexed tables, called Subgraphs, allowing any application  
+to perform a query to its protocol and receive the response.  
+
+The Graph comes with its own crypto currency called GRT.
+
+its protocol include 3 types of actors:
+
+Curators: They are subgraph developers who assess which subgraphs need to be indexed by The Graph.  
+They needs to attach GRT to these subgraphs.    
+
+Indexers: Node operators tasked with providing indexing and querying services for the signaled subgraphs,  
+and must stake GRT in order to provide these services.  
+
+Delegators: Delegate GRT to indexers in order to contribute to running the network without installing a node.  
+
+## 2 Install the Graph 
+The Graph CLI is written in JavaScript and you will need to have either npm or yarn installed to use it.
+
+# NPM
+$ npm install -g @graphprotocol/graph-cli
+
+# Yarn
+$ yarn global add @graphprotocol/graph-cli
+
+## 3 Creating a subgraph
+
+For creating a subgraph, you have 2 options.  
+If it is on mainnet you can create it using The Graph studio.  
+
+For testnets other environnements (and mainnet) we must use hosted service.  
+This is the option we are going to take in this tutorial.  
+
+When creating our subgraph, we can either create it from an example and then customize it:  
+
+```shell
+graph init --product hosted-service --from-example
+```
+
+Or we can create it from a contract.
+In this example we will create it from the faucet contract (at 0x8e19Ca559eAB83a5201663E8a70cd0C2A9479663) on the Rinkeby network:  
+
+```shell
+yarn graph init --product hosted-service --from-contract 0x8e19Ca559eAB83a5201663E8a70cd0C2A9479663
+```
+
+### 3.1 setting subgraph name
+
+You will be asked to provide a subgraph name.
+The require format is 
+
+```shell
+<GITHUB_NAME>/<GITHUB_REPOSITORY>
+```
+
+![TheGraph1](../../img/thegraph/TheGraph1.png "Step 1")
```

Comment:
I don't think all those images are useful, maybe one

Reply:
Yes some pictures can be removed.